builtin/providers/spotinst/resource_spotinst_aws_group.go

```diff
@@ -0,0 +1,2077 @@
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+
+			"ec2_container_service_integration": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"cluster_name": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+
+			"kubernetes_integration": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"api_server": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"token": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func scalingPolicySchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeSet,
+		Optional: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"policy_name": &schema.Schema{
+					Type:     schema.TypeString,
+					Required: true,
+				},
+
+				"metric_name": &schema.Schema{
+					Type:     schema.TypeString,
+					Required: true,
+				},
+
+				"statistic": &schema.Schema{
+					Type:     schema.TypeString,
+					Required: true,
+				},
+
+				"unit": &schema.Schema{
+					Type:     schema.TypeString,
+					Required: true,
+				},
+
+				"threshold": &schema.Schema{
+					Type:     schema.TypeFloat,
+					Required: true,
+				},
+
+				"adjustment": &schema.Schema{
+					Type:     schema.TypeInt,
+					Optional: true,
+				},
+
+				"min_target_capacity": &schema.Schema{
+					Type:     schema.TypeInt,
+					Optional: true,
+				},
+
+				"max_target_capacity": &schema.Schema{
+					Type:     schema.TypeInt,
+					Optional: true,
+				},
+
+				"namespace": &schema.Schema{
+					Type:     schema.TypeString,
+					Required: true,
+				},
+
+				"operator": &schema.Schema{
+					Type:     schema.TypeString,
+					Optional: true,
+				},
+
+				"evaluation_periods": &schema.Schema{
+					Type:     schema.TypeInt,
+					Required: true,
+				},
+
+				"period": &schema.Schema{
+					Type:     schema.TypeInt,
+					Required: true,
+				},
+
+				"cooldown": &schema.Schema{
+					Type:     schema.TypeInt,
+					Required: true,
+				},
+
+				"dimensions": &schema.Schema{
+					Type:     schema.TypeMap,
+					Optional: true,
+					Elem:     &schema.Schema{Type: schema.TypeString},
+				},
+			},
+		},
+	}
+}
+
+func resourceSpotinstAwsGroupCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*spotinst.Client)
+	newAwsGroup, err := buildAwsGroupOpts(d, meta)
+	if err != nil {
+		return err
+	}
+	log.Printf("[DEBUG] AwsGroup create configuration: %#v\n", newAwsGroup)
+	res, _, err := client.AwsGroup.Create(newAwsGroup)
+	if err != nil || len(res) == 0 {
+		return fmt.Errorf("[ERROR] Error creating group: %s", err)
+	}
+	d.SetId(*res[0].ID)
```

Comment:
The Spotinst API returns a response in the following format:
```json
"request": {
  "id": "e7f40fb2-cbf0-4f47-8d9c-7031668953d8",
  "url": "/aws/ec2/group/<group_id>",
  "method": "GET",
  "timestamp": "2016-10-31T07:39:58.858Z"
},
"response": {
  "status": {
    "code": 200,
    "message": "OK"
  },
  "kind": "spotinst:aws:ec2:group",
  "items": [
    {
      ...
    }
  ],
  "count": 1
}
```
On successful group creation, the `items` always contains only one item. But just to be on the safe side, I'll add a check for that.